Data/HashMap/Internal/Array.hs

```diff
@@ -474,6 +478,27 @@ fromList n xs0 =
     go (x:xs) mary i = do write mary i x
                           go xs mary (i+1)
 
+fromList' :: Int -> [a] -> Array a
+fromList' n xs0 =
+    CHECK_EQ("fromList'", n, Prelude.length xs0)
+        run $ do
+            mary <- new_ n
+            go xs0 mary 0
+  where
+    go [] !mary !_   = return mary
+    go (!x:xs) mary i = do write mary i x
+                           go xs mary (i+1)
+
+instance TH.Lift a => TH.Lift (Array a) where
+#if MIN_VERSION_template_haskell(2,16,0)
+  liftTyped ar = [|| fromList' arlen arlist ||]
```

Comment:
It's disappointing that `SmallArray#` (and other `*Array#`) don't have literals, but I don't think we can do better atm.

Reply:
Fortunately, I don't that any overhead converting the arrays from lists will really be noticeable, especially since I expect all this spliced stuff gets floated out as CAFs. Literals would be nice.

Reply:
they will be CAFs, but they could been off-heap structures, which is even better.

Reply:
See https://well-typed.com/blog/2020/06/th-for-static-data/#disclaimer-this-wont-work-for-all-data-types

Reply:
You mean just in the program code? Do we have things with pointers that work like that?

Reply:
E.g. `Map` and `IntSet`, yes.